lib/elixir/lib/enum.ex

```diff
@@ -1724,6 +1736,14 @@ defmodule Enum do
                       fn x, acc -> {:cont, fun.(x, acc)} end) |> elem(1)
   end
 
+  defp reduce_handling_empty(enumerable, fun, empty_fallback) do
+    if Enum.empty?(enumerable) do
```

Comment:
We cannot call `Enum.empty?` because it may force the enumerable to run twice. We need to use the same schema we used in `Enum.min_max`.